### DISCUSSION TITLE
Github Copilot code formatting

### DISCUSSION THREAD

**Discussion by leo-3D** [community/community#52859] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Environment:
Microsoft Visual Studio 2022 (64-bit) - Current Version 17.5.4
GitHub Copilot 4/14/2023 1.82.0.0

Is it me (user error) or toher folks see strange indenting too?
![copilot 1](https://user-images.githubusercontent.com/27030396/232152905-459621df-02e9-4451-94ea-0d88f43940b5.png)
TAB results in 
![coppilot 2](https://user-images.githubusercontent.com/27030396/232152968-13fab6ab-6066-42a0-9f4d-99d0075ff00f.png)
where:
![indenting](https://user-images.githubusercontent.com/27030396/232153017-f4b69fc8-ac27-431b-92e3-3f8080caded8.png)

**Comment by CodyALohse** [comment]
I also see this issue:

Visual Studio 2022 17.5.5
Copilot: 1.82.1.0

![indentIssue](https://github.com/community/community/assets/12501808/a04159e1-6aa3-4cbe-8949-48bbaee8c813)

<img width="385" alt="image" src="https://github.com/community/community/assets/12501808/43442f3d-6be2-4fb5-aaca-933e13f46cfb">

**Reply by Akash1134** [reply]
Hi @CodyALohse , Thanks for reporting this issue. Have you been facing it quite often or this is just a one off incident? Does it happen only with C/C++ or have you experienced it with other languages in Visual Studio 2022?

**Reply by CodyALohse** [reply]
It's fairly consistent. I'm using C# and I haven't tried to reproduce with other languages. I've attached a minimal reproduceable (at least reproducible for me) VS solution. I hope that's helpful.

I'm using ctrl + alt + \ to trigger copilot in this example:

![devenv_aYmxdTFcHL](https://github.com/community/community/assets/12501808/ddb93494-16b1-463f-a0eb-d9ee96da4b98)

[TestProject1.zip](https://github.com/community/community/files/11442482/TestProject1.zip)

**Comment by pontusnaslund** [comment, marked as answer]
I also had this issue but it appears to have been fixed for me with VS2022 Version 17.6.0 Preview 6.0

**Comment by Akash1134** [comment]
Thank You @pontusnaslund for sharing. @leo-zspace and @CodyALohse - could you please check if updating to the latest version of VS2022 Version 17.6.0 Preview 6.0 solves the issue for you?

**Reply by CodyALohse** [reply]
Yes, it appears to be resolved in the latest preview build (17.6.0 Preview 7.0).

**Comment by janlugt** [comment]
I'm using Visual Studio 2022 17.6.2, and I'm still seeing this issue. Did any of the fixes from the previews not make it in the regular build?

<details>
  <summary>Full version info</summary>
Microsoft Visual Studio Professional 2022
Version 17.6.2
VisualStudio.17.Release/17.6.2+33723.286
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Professional

ASP.NET and Web Tools   17.6.326.62524
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.6.326.62524
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.6.326.62524
Azure Functions and Web Jobs Tools

C# Tools   4.6.0-3.23259.8+c3cc1d0ceeab1a65da0217e403851a1e8a30086a
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitHub Copilot   1.84.0.1 (v1.84.0.1@760da3695)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.84.51 (v1.84.0)

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.6.0.2326105+37f2727000f6a964584cafb86e69cf0b39baecb7
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.6.13.0
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.20329.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.6.0-3.23259.8+c3cc1d0ceeab1a65da0217e403851a1e8a30086a
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.6.0-beta.23174.5+0207bea1afae48d9351ac26fb51afc8260de0a97
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.
</details>

**Comment by LaRoomy** [comment]
I'm using Visual Studio 17.6.3 and facing the same issue. I'm using C/C++. A fix is to change the tab setting to 'insert spaces'. If this is done it works as expected, but I don't want that.

**Comment by nbarbettini** [comment]
This is still an issue on Visual Sutdio 17.6.4 and Github Copilot 1.84.0.1 (latest versions of both).

editorconfig settings:

```
[*]
end_of_line = crlf
insert_final_newline = true
charset = utf-8
tab_width = 4
indent_size = 2

# Csharp files
[*.cs]

#### Core EditorConfig Options ####

# Indentation and spacing
indent_size = 4
indent_style = space
tab_width = 4
```

VS settings:
![image](https://github.com/community/community/assets/7525482/bba17f8d-32f2-4eb1-87e8-33b1fb22999a)

I **turned off** adaptive formatting and this issue seemed to occur less, but it didn't completely go away.

**Comment by sethflowers** [comment]
I have this issue under VS 17.7.4 with C#. Changing from "Smart" to "Block" does nothing.

![image](https://github.com/community/community/assets/3913676/07d6f83e-0be8-4f21-9f40-51314eb2a724)

![image](https://github.com/community/community/assets/3913676/8e11bb53-9ff1-4bb4-90b9-f25bcd6a7ba4)

**Comment by LockTar** [comment]
Using 17.7.5 and the issue is still there... Everything latest version

**Comment by parched** [comment]
I'm running 17.8.1 and it's still present, a colleague of mine is running the latest preview and says it's still present there too.

Is there any road map for a fix @Akash1134, do you need any more reproduction information?

**Comment by AntonZhuchkovXOR** [comment]
I just upgraded to Version 17.8.5 and the issue is still there. Can you fix it please, it really impairs a user experience with CoPilot.

**Comment by guillaume86** [comment]
Issue is still there.

**Comment by ayim** [comment]
@guillaume86 @AntonZhuchkovXOR @parched @LockTar we've got a pull request in that may address the issue.
I will check in again here once the fix is released and ready for y'all to try.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio

**Reply by LockTar** [reply]
Hi,
Any news about the PR?
